src/java/org/apache/cassandra/utils/FilterFactory.java

```diff
@@ -90,7 +95,14 @@ private static IFilter createFilter(int hash, long numElements, int bucketsPer,
             logger.error("Failed to create new Bloom filter with {} elements: ({}) - " +
                          "continuing but this will have severe performance implications, consider increasing FP chance or" +
                          "lowering number of sstables through compaction", numElements, e.getMessage());
+            FilterFactoryMetrics.oomCounter.inc();
             return AlwaysPresent;
         }
     }
+
+    static final class FilterFactoryMetrics
+    {
+        private static final MetricNameFactory metricNameFactory = new DefaultNameFactory("BloomFilter");
+        private static final Counter oomCounter = Metrics.counter(metricNameFactory.createMetricName("OutOfMemory")); // counting how many times we failed to load BloomFilter due to not enough memory
```

Comment:
how does this metric map to CNDB metrics in Prometheus ?

Reply:
Good point. If we want it in Prometheus then different API should be used. Let me rework this.